src/stcal/ramp_fitting/tests/test_gls_fit.py

```diff
@@ -0,0 +1,922 @@
+import copy
```

Comment:
Generally, for these tests, how/where are gain & readnoise included in the test ramp data to be fit  ?

Also, I assume that we will go through another review iteration sometime, once the tests that are now to be skipped are enabled.

Reply:
The tests as currently written were essentially copied from the JWST class that originally was supposed to test ordinary least squares and generalized least squares.  This was done in a class at:

https://github.com/spacetelescope/jwst/blob/8aff8ed5f3303c87b1ee721fa27b40faac6f57ec/jwst/ramp_fitting/tests/test_ramp_fit.py#L218

Many of these tests simply hard coded out the GLS algorithm by hard coding "OLS" as the algorithm to be run, which simply ran the tests twice using the same algorithm.  I used these test as a guide to find bugs and deficiencies existing in the GLS code.  

Yes you are correct about later iterations of the code.  As these bugs/deficiencies get fixed the currently skipped/failed tests will be enabled or modified to be enabled.  Right now they are place holders to note problems with the existing GLS code.